app/components/autocomplete/channel_mention/index.ts

```diff
@@ -79,8 +85,23 @@ const withTeamId = withObservables(['teamId', 'channelId'], ({teamId, channelId,
     };
 });
 
-const enhanced = withObservables(['value', 'isSearch', 'teamId', 'cursorPosition'], ({value, isSearch, teamId, cursorPosition, database}: OwnProps) => {
-    const matchTerm = getMatchTermForChannelMention(value.substring(0, cursorPosition), isSearch);
+const withMatchPattern = withObservables(['isSearch'], ({isSearch, database}: WithMatchProps) => {
```

Comment:
this is fine, but I would ask if you can remove the extra enhancer and maybe get the `matchTerm` as an observable that you can pipe to get the `localChannels`. My only reason for this is to avoid having `matchPattern` as a prop of the component

Reply:
Oh okay, so move this logic to `enhanced` then rather than having it in a separate enhancer? I don't entirely know why I decided I didn't want to access the database in `enhanced` to get the config value there

Reply:
This should be fixed now